app/services/FrontendSessionService.scala

```diff
@@ -177,4 +179,42 @@ class FrontendSessionService @Inject()(val sessionCache: FrontendSessionsReposit
     } else {
       Some(SAVED_STATUS)
     }
+
+  def fetchPartFromCompanyDetailsList[A](index: Int)(implicit request: Request[_], formats: json.Format[A]): Future[Option[A]] = {
+    sessionCache.getFromSession[JsValue](DataKey(SUBSIDIARY_COMPANIES_CACHE)).map {
+      subsidiaryCompanies =>
+        subsidiaryCompanies.map(_.\(COMPANIES).as[JsArray].\(index).getOrElse(Json.obj()).as[A])
+    } recover {
+      case x: Throwable => {
+        logger.debug("[ERSUtil][fetchPartFromCompanyDetailsList] Nothing found in cache, expected if this is not an edit journey: " + x.getMessage)
```

Comment:
After deliberation I feel that it could be changed however will be a net increase in the lines of code for another check and the recover block would still be needed to handle other error scenarios.

Reply:
I see your point, although if it is a genuine error then catching any `Throwable` just to log at debug level is arguably insufficient anyway.  If the same pattern is used elsewhere it might be better to handle as technical debt in future.